### ISSUE TITLE
OSX backend crashes processing keyboard events when rendering from DisplayLink

### ISSUE THREAD

**Issue by TheMostDiligent** [ocornut/imgui#5940]
**Version/Branch of Dear ImGui:**

Version: 1.89.1
Branch: master

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_osx.mm + [custom](https://github.com/DiligentGraphics/DiligentTools/tree/master/Imgui/src)
Compiler: AppleClang 13.1.6.13160021 (XCode 13.4)
Operating System: MacOS (M1, arm64)

**My Issue/Question:**

When using DisplayLink, render commands are called from a separate thread that does not allow processing input events. In current implementation, however, `ImGui::Render` calls `ImGui::EndFrame` that eventually calls [this callback](https://github.com/ocornut/imgui/blob/bf4c2e00c09384a56fa1a4df30443e56ebcb85fe/backends/imgui_impl_osx.mm#L460) which results in crash:

```cpp
    io.SetPlatformImeDataFn = [](ImGuiViewport* viewport, ImGuiPlatformImeData* data) -> void
    {
        ImGui_ImplOSX_Data* bd = ImGui_ImplOSX_GetBackendData();
        if (data->WantVisible)
        {
            [bd->InputContext activate]; <- Crash here
        }
        else
        {
            [bd->InputContext discardMarkedText];
            [bd->InputContext invalidateCharacterCoordinates];
            [bd->InputContext deactivate];
        }
        [bd->KeyEventResponder setImePosX:data->InputPos.x imePosY:data->InputPos.y + data->InputLineHeight];
    };
```

As an additional note, it would be really nice if there still was an option to handle OSX events manually.

**Comment by TheMostDiligent** [comment]
> A PR/fix would be welcome for the crash.

I haven't updated dear imgui for a while and I see that *a lot* has changed in OSX implementation. I don't know the reasons, but it appears that overall changes were going in the wrong direction: input event processing is now completely hidden from the application (unlike Windows and Linux where it is explicit), which has no control over when it is done. Like I wrote in my issue, this design is incompatible with the Display Link approach (which is in fact recommended by Apple), where rendering and input processing are decoupled.
I don't see how this can be easily fixed without making `ImGui_ImplOSX_HandleEvent` public again, so that the application can decide which thread and at what point this should be called from.

**Comment by ocornut** [comment]
You can use git log and blame to find the reasons, if any.
Again, this code has no explicit maintainer, I have no idea what DisplayLink is, so it'll need someone to actually dig and devise solutions, while keeping in mind other use cases. That or, you use a portable backend.

**Comment by TheMostDiligent** [comment]
Yes, I did use git log to see where the changes happened (so far I am testing 1.85, but apparently will need to go further back due to other issues).

> That or, you use a portable backend.

That appears to be a more feasible approach at this moment.
